src/java.base/share/classes/jdk/internal/math/DoubleToDecimal.java

```diff
@@ -149,71 +156,42 @@ public static String toString(double v) {
      * @param fd    the object that will carry <i>f</i>, <i>e</i>, and <i>n</i>.
      */
     public static void split(double v, FormattedFPDecimal fd) {
-        new DoubleToDecimal(true).toDecimal(v, fd);
+        byte[] str = new byte[MAX_CHARS];
+        LATIN1.toDecimal(str, 0, v, fd);
     }
 
     /**
-     * Appends the rendering of the {@code v} to {@code app}.
+     * Appends the rendering of the {@code v} to {@code str}.
      *
      * <p>The outcome is the same as if {@code v} were first
      * {@link #toString(double) rendered} and the resulting string were then
-     * {@link Appendable#append(CharSequence) appended} to {@code app}.
      *
-     * @param v the {@code double} whose rendering is appended.
-     * @param app the {@link Appendable} to append to.
+     * @param str the String byte array to append to
+     * @param index the index into str
+     * @param v the {@code double} whose rendering is into str.
      * @throws IOException If an I/O error occurs
      */
-    public static Appendable appendTo(double v, Appendable app)
-            throws IOException {
-        return new DoubleToDecimal(false).appendDecimalTo(v, app);
-    }
+    public int putDecimal(byte[] str, int index, double v) {
+        assert index >= 0 && index + MAX_CHARS <= length(str) : "Trusted caller missed bounds check";
```

Comment:
```suggestion
        assert 0 <= index && index <= length(str) - MAX_CHARS : "Trusted caller missed bounds check";
```
This avoids a potential overflow.
But if you want to enforce the check, you should probably avoid using `assert` since assertions might be disabled.